File: Blog.Site/Scripts/react/Src/Types/CommentDTO.ts
import RecordDTO from "types/RecordDTO";
import UserDTO from "types/UserDTO";

export default class CommentDTO {
  public CommentId: number;

  public Content: string;

  public CreateDate: string;

  public RecordId: number;

  public Record: RecordDTO;

  public UserId: number;

  public User: UserDTO;
}

File: Blog.Services/Models/CommentDTO.cs
using Blog.Services.Identity.Models;
using System;
using System.ComponentModel.DataAnnotations;

namespace Blog.Services.Models
{
    public class CommentDTO
    {
        public int CommentId { get; set; }

        [Required]
        public string Content { get; set; }

        public DateTime CreateDate { get; set; }

        public int RecordId { get; set; }

        public RecordDTO Record { get; set; }

        public int UserId { get; set; }

        public UserDTO User { get; set; }
    }
}
